pkg/reconciliation/reconcile_fql.go

```
@@ -0,0 +1,100 @@
+// To ease integration into the reconciliation process, it returns (shouldFQLBeEnabled, serverMajorVersion, ReconcileResult) where the ReconcileResult may be result.Error() or result.Continue().
+func parseFQLFromConfig(rc *ReconciliationContext) (bool, int64, result.ReconcileResult) {
+	dc := rc.GetDatacenter()
+	serverMajorVersion, err := strconv.ParseInt(strings.Split(dc.Spec.ServerVersion, ".")[0], 10, 8)
+	if err != nil {
+		rc.ReqLogger.Error(err, "error parsing server major version. Can't enable full query logging without knowing this")
+		return false, serverMajorVersion, result.Error(err)
+	}
+	shouldFQLBeEnabled := false
+	if dc.Spec.Config != nil {
+		var dcConfig map[string]interface{}
+		if err := json.Unmarshal(dc.Spec.Config, &dcConfig); err != nil {
+			rc.ReqLogger.Error(err, "error unmarshalling DC config JSON")
+			return false, serverMajorVersion, result.Error(err)
+		}
+		casYaml, found := dcConfig["cassandra-yaml"]
+		if !found {
+			return false, serverMajorVersion, result.Continue()
+		}
+		casYamlMap, ok := casYaml.(map[string]interface{})
+		if !ok {
+			err := fmt.Errorf("type casting error")
+			rc.ReqLogger.Error(err, "couldn't cast cassandra-yaml value from config to map[string]interface{}")
+			return false, serverMajorVersion, result.Error(err)
+		}
+		if _, found := casYamlMap["full_query_logging_options"]; found {
+			if serverMajorVersion < 4 || dc.Spec.ServerType != "cassandra" {
+				err := fmt.Errorf("full query logging only supported on OSS Cassandra 4x+")
+				rc.ReqLogger.Error(err, "full_query_logging_options is defined in Cassandra config, it is not supported on the version of Cassandra you are running")
+				return false, serverMajorVersion, result.Error(err)
+			}
+			rc.ReqLogger.Info("full_query_logging_options is defined in Cassandra config, we will try to enable it via the management API")
+			shouldFQLBeEnabled = true
+		}
+	}
+	if dc.Spec.ServerType != "cassandra" {
```

Shouldn't this be before all the other logic? Unnecessary to do any work if the server is not ``cassandra``

---

In your original patch it was, however we need to determine whether FQL is turned on in the `spec.config` and throw an error if so. Otherwise DSE clusters trying to use FQL will just silently fail to enable it.